packages/catalog-server/src/lib/graphql.ts

```diff
@@ -0,0 +1,127 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {readFile} from 'fs/promises';
+import {createRequire} from 'module';
+import {makeExecutableSchema} from '@graphql-tools/schema';
+import {isReadablePackage, isReadablePackageVersion, PackageInfo, Resolvers} from '@webcomponents/catalog-api/lib/schema.js';
+import {Catalog} from './catalog.js';
+
+const require = createRequire(import.meta.url);
+
+/**
+ * Creates a GraphQLSchema object with resolvers
+ */
+export const makeExecutableCatalogSchema = async (catalog: Catalog) => {
+  const schemaPath = require.resolve(
+    '@webcomponents/catalog-api/src/lib/schema.graphql'
+  );
+  const schemaSource = await readFile(schemaPath, 'utf8');
+
+  // Resolvers for our schema. See https://graphql.org/learn/execution/ for
+  // an explaination of the role of resolvers in performing GraphQL queries.
+  const resolvers: Resolvers = {
+    Query: {
```

Comment:
So `Query` is a special thing meaning the entrypoint query. Do we only ever have 1 entrypoint query? It seems like this resolver is assuming that all queries include a packageName, but we'll also support lookups by tag name, right?

Reply:
Query and Mutation are special "root" types. Right now we're only implementing the `package` "field" (which is also a query). We will add more queries to the schema and resolves as we need them. The `elements()` query that's in the schema (and marked as not implemented) will be one of the main queries. This query is for accessing specific packages/versions/custom-elements in a top-down manner. `elements()` will be for actually searching across the whole catalog.